### PR TITLE
adding build.json template file

### PR DESCRIPTION
**Pull request by laurafitzgerald**
**JIRA**
https://issues.jboss.org/browse/RHMAP-15733

@odra

### CODE REVIEW

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 1]
```diff
@@ -1,14 +1,72 @@
...
-node('android') {
-   stage 'Checkout'
-	 checkout scm
 
-	 stage 'Prepare'
-	 cordova platform add android
-	 cordova prepare
+def platform = params?.PLATFORM?.trim()
+def buildType = params?.BUILD_CONFIG?.trim()
+def keystore_path = "/opt/android-sdk-linux/android.debug"
```
Comment (odra) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: I think we can use the ANDROID_HOME env var in this string: `"${env.ANDROID_HOME}/android.debug"`

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 2]
```diff
@@ -1,14 +1,72 @@
...
 
-	 stage 'Build'
-	 cordova build android
+if(platform == 'android') {
```
Comment (odra) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: can we move this platform check into the stage blocks when its required (prepare and archive)?

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 3]
```diff
@@ -1,14 +1,72 @@
...
+            if (buildType == 'release') {
+                archiveArtifacts artifacts: 'platforms/android/build/outputs/apk/android-release.apk', excludes: 'platforms/android/build/outputs/apk/*-unaligned.apk'
+            } else {
+                archiveArtifacts artifacts: 'platforms/android/build/outputs/apk/android-debug.apk', excludes: 'platforms/android/build/outputs/apk/*-unaligned.apk'
+            }
+        }
+    }
+}
+
+if (platform == 'ios') {
```
Comment (odra) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: I think we can remove this block

Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: @odra Do you mean remove the entire ios steps?

Comment (odra) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: yes, I was thinking in moving the platform verification part to the stages that need this verification (prepare and archive)

**Comment by laurafitzgerald** [comment]
@odra made the suggested change if you could take a look again :)

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 4]
```diff
@@ -1,14 +1,55 @@
...
+	            file_content= sh (
+	                script: 'jq --arg test_path /opt/android-sdk-linux/android.debug \'.android.debug.keystore = $test_path\' -r build.json',
+	                returnStdout: true
+	                )
+	        	sh "echo '${file_content}' > build.json"
+	        }else{
+	            file_content= sh (
+	                script: "jq --arg test_path '${keystore_path}' \'.android.debug.keystore = $test_path\' -r build.json",
+	                returnStdout: true
+	                )
```
Comment (wei-lee) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: how does this work for the release mode?

Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: @wei-lee yep that should be android.release.keystore. will fix

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 5]
```diff
@@ -1,14 +1,55 @@
...
+	        	sh "echo '${file_content}' > build.json"
+	        }else{
+	            file_content= sh (
+	                script: "jq --arg test_path '${keystore_path}' \'.android.release.keystore = $test_path\' -r build.json",
+	                returnStdout: true
+	                )
+		        sh "echo '${file_content}' > build.json"
```
Comment (wei-lee) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: I am not sure this is the right way to do it.

For debug build, the default keystore should be ok and I don't think we need to override.
 
For release builds. When building for release, each build will create its own keystore and it's saved in Jenkins credentials.  I think it's best not to sign the app here, but use the signApk plugin to sign the app, as we did in Android native app.

Is it possible to not sign the app when run `cordova build`?

Comment (odra) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: I think it will generate an unsigned apk if we run "cordova build android --release"

Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: @wei-lee Where does the keystore path get stored? In some param that we can access? @odra In the case of an unsigned debug app do we not pass the build.json file?

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 6]
```diff
@@ -1,14 +1,64 @@
...
+	            apksToSign: "platforms/android/**/*-unsigned.apk",
+	            // uncomment the following line to output the signed APK to a separate directory as described above
+	            // signedApkMapping: [ $class: UnsignedApkBuilderDirMapping ],
+	            // uncomment the following line to output the signed APK as a sibling of the unsigned APK, as described above, or just omit signedApkMapping
+	            // you can override these within the script if necessary
+	            // androidHome: '/usr/local/Cellar/android-sdk'
+	        )
+		    } else {
+		      println('Debug Build - Using default developer signing key')
+		    }
```
Comment (wei-lee) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: indentation looks bit off here

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 7]
```diff
@@ -1,14 +1,64 @@
...
+    	if (platform == 'android') {
+    		if (buildType == 'debug') {
+        		sh 'cordova build $platform --buildConfig build.json'
+        	} else {
+        		sh 'cordova build $platform --release'
+        	}
+        }
+    }
+
+    stage("Sign") {
```
Comment (wei-lee) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: I guess we are still blocked by @odra's work to sign ios apps, right?

Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: @wei-lee yes the android part is now complete so we can merge it. And as suggested do the iOS piece under a different ticket.

File: jenkinsfiles/cordova_jenkinsfile.groovy (hunk 6 above)
Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: @wei-lee yes it's a strange one. It's correct locally. I will check out how to fix this now.

Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: and same below

File: jenkinsfiles/cordova_jenkinsfile.groovy (hunk 7 above)
Comment (odra) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: Correct, I will update the build.json file once the plugin support the keychain as a pipeline step.

**Comment by laurafitzgerald** [comment]
@odra @wei-lee format fixed and commits squashed. Final review before merging?

File: jenkinsfiles/cordova_jenkinsfile.groovy [hunk 8]
```diff
@@ -1,14 +1,63 @@
...
+        checkout scm
+    }
+
+    stage("Prepare") {
+        sh "cordova platform add $platform"
+        sh "cordova prepare $platform"
+    }
+
+    stage("Build") {
+		if (platform == 'android') {
```
Comment (wei-lee) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: indentation here...

Comment (laurafitzgerald) [inline jenkinsfiles/cordova_jenkinsfile.groovy]: :see_no_evil: fixing

**Pull request merged**